### ISSUE TITLE
etcd restore fails on a single node cluster

### ISSUE THREAD

**Issue by soumyalj** [rancher/rancher#17803]
Rancher Version: Master

Steps:

1. Create a single node DO cluster with all 3 roles.
2. Create few workloads
3. Perform a backup
4. Restore from the backup

Restore fails and the cluster is stuck in "Updating" state

```
This cluster is currently Updating.

Cluster must have at least one etcd plane host: failed to connect to the following etcd host(s) [abcd]


```

Server logs below:

```
019/02/01 01:07:17 [ERROR] Failed to update cluster [c-mc5hr]: Operation cannot be fulfilled on clusters.management.cattle.io "c-mc5hr": the object has been modified; please apply your changes to the latest version and try again
2019/02/01 01:07:18 [INFO] Updating workload [ingress-nginx/nginx-ingress-controller] with public endpoints [[{"nodeName":"c-mc5hr:m-r6ltd","addresses":[""],"port":80,"protocol":"TCP","podName":"ingress-nginx:nginx-ingress-controller-sz4mc","allNodes":false},{"nodeName":"c-mc5hr:m-r6ltd","addresses":["],"port":443,"protocol":"TCP","podName":"ingress-nginx:nginx-ingress-controller-sz4mc","allNodes":false}]]
2019/02/01 01:07:22 [INFO] [healthcheck] service [kube-apiserver] on host [abcd] is healthy
2019/02/01 01:07:23 [INFO] [controlplane] Successfully started [rke-log-linker] container on host [abcd]
2019/02/01 01:07:23 [INFO] [remove/rke-log-linker] Successfully removed container on host [abcd]
2019/02/01 01:07:23 [INFO] [controlplane] Successfully started [kube-controller-manager] container on host [abcd]
2019/02/01 01:07:23 [INFO] [healthcheck] Start Healthcheck on service [kube-controller-manager] on host
2019/02/01 01:07:28 [INFO] [healthcheck] service [kube-controller-manager] on host [abcd] is healthy
2019/02/01 01:07:29 [INFO] [controlplane] Successfully started [rke-log-linker] container on host 
2019/02/01 01:07:29 [INFO] [remove/rke-log-linker] Successfully removed container on host [abcd]
2019/02/01 01:07:29 [INFO] [controlplane] Successfully started [kube-scheduler] container on host [abcd]
2019/02/01 01:07:29 [INFO] [healthcheck] Start Healthcheck on service [kube-scheduler] on host [abcd]

2019/02/01 01:07:34 [INFO] [healthcheck] service [kube-scheduler] on host [abcd] is healthy
2019/02/01 01:07:34 [INFO] [controlplane] Successfully started [rke-log-linker] container on host [abcd]
2019/02/01 01:07:35 [INFO] [remove/rke-log-linker] Successfully removed container on host [abcd]
2019/02/01 01:07:35 [INFO] [controlplane] Successfully started Controller Plane..
2019/02/01 01:07:35 [INFO] [authz] Creating rke-job-deployer ServiceAccount
2019/02/01 01:07:35 [INFO] [authz] rke-job-deployer ServiceAccount created successfully
2019/02/01 01:07:35 [INFO] [authz] Creating system:node ClusterRoleBinding
2019/02/01 01:07:35 [INFO] [authz] system:node ClusterRoleBinding created successfully
2019/02/01 01:07:35 [INFO] Successfully Deployed state file at [management-state/rke/rke-104494747/cluster.rkestate]
2019/02/01 01:07:35 [INFO] [state] Saving full cluster state to Kubernetes
2019/02/01 01:07:35 [INFO] [state] Successfully Saved full cluster state to Kubernetes ConfigMap: cluster-state
2019/02/01 01:07:35 [INFO] [worker] Building up Worker Plane..
2019/02/01 01:07:35 [INFO] [sidekick] Sidekick container already created on host [abcd]
2019/02/01 01:07:35 [INFO] [worker] Successfully started [kubelet] container on host 
2019/02/01 01:07:35 [INFO] [healthcheck] Start Healthcheck on service [kubelet] on host 
2019/02/01 01:07:40 [INFO] error in remotedialer server [400]: websocket: close 1006 unexpected EOF
2019/02/01 01:07:42 [INFO] Updating workload [ingress-nginx/nginx-ingress-controller] with public endpoints [null]
2019/02/01 01:07:43 [INFO] Handling backend connection request [c-mc5hr:m-r6ltd]
2019/02/01 01:07:45 [INFO] Updating workload [ingress-nginx/nginx-ingress-controller] with public endpoints [[{"nodeName":"c-mc5hr:m-r6ltd","addresses":,"port":80,"protocol":"TCP","podName":"ingress-nginx:nginx-ingress-controller-sz4mc","allNodes":false},{"nodeName":"c-mc5hr:m-r6ltd","addresses":,"port":443,"protocol":"TCP","podName":"ingress-nginx:nginx-ingress-controller-sz4mc","allNodes":false}]]
2019/02/01 01:07:45 [INFO] [healthcheck] service [kubelet] on host [1 is healthy
2019/02/01 01:07:46 [INFO] [worker] Successfully started [rke-log-linker] container on host 
2019/02/01 01:07:46 [INFO] [remove/rke-log-linker] Successfully removed container on host 
2019/02/01 01:07:46 [INFO] [worker] Successfully started [kube-proxy] container on host 
2019/02/01 01:07:46 [INFO] [healthcheck] Start Healthcheck on service [kube-proxy] on host 
2019/02/01 01:07:51 [INFO] [healthcheck] service [kube-proxy] on host [abcd] is healthy
2019/02/01 01:07:52 [INFO] [worker] Successfully started [rke-log-linker] container on host 
2019/02/01 01:07:52 [INFO] [remove/rke-log-linker] Successfully removed container on host 
2019/02/01 01:07:52 [INFO] [worker] Successfully started Worker Plane..
2019/02/01 01:07:53 [INFO] [cleanup] Successfully started [rke-log-cleaner] container on host 
2019/02/01 01:07:53 [INFO] [remove/rke-log-cleaner] Successfully removed container on host 
2019/02/01 01:07:53 [INFO] [sync] Syncing nodes Labels and Taints
2019/02/01 01:07:53 [INFO] [sync] Successfully synced nodes Labels and Taints
2019/02/01 01:07:53 [INFO] [network] Setting up network plugin: canal
2019/02/01 01:07:53 [INFO] [addons] Saving ConfigMap for addon rke-network-plugin to Kubernetes
2019/02/01 01:07:53 [INFO] [addons] Successfully saved ConfigMap for addon rke-network-plugin to Kubernetes
2019/02/01 01:07:53 [INFO] [addons] Executing deploy job rke-network-plugin
2019/02/01 01:07:53 [INFO] [addons] Setting up kubedns
2019/02/01 01:07:53 [INFO] [addons] Saving ConfigMap for addon rke-kubedns-addon to Kubernetes
2019/02/01 01:07:53 [INFO] [addons] Successfully saved ConfigMap for addon rke-kubedns-addon to Kubernetes
2019/02/01 01:07:53 [INFO] [addons] Executing deploy job rke-kubedns-addon
2019/02/01 01:07:53 [INFO] [addons] kubedns deployed successfully
2019/02/01 01:07:53 [INFO] [dns] DNS provider kubedns deployed successfully
2019/02/01 01:07:53 [INFO] [addons] Setting up Metrics Server
2019/02/01 01:07:53 [INFO] [addons] Saving ConfigMap for addon rke-metrics-addon to Kubernetes
2019/02/01 01:07:53 [INFO] [addons] Successfully saved ConfigMap for addon rke-metrics-addon to Kubernetes
2019/02/01 01:07:53 [INFO] [addons] Executing deploy job rke-metrics-addon
2019/02/01 01:07:53 [INFO] [addons] Metrics Server deployed successfully
2019/02/01 01:07:53 [INFO] [ingress] Setting up nginx ingress controller
2019/02/01 01:07:53 [INFO] [addons] Saving ConfigMap for addon rke-ingress-controller to Kubernetes
2019/02/01 01:07:53 [INFO] [addons] Successfully saved ConfigMap for addon rke-ingress-controller to Kubernetes
2019/02/01 01:07:53 [INFO] [addons] Executing deploy job rke-ingress-controller
2019/02/01 01:07:53 [INFO] [ingress] ingress controller nginx deployed successfully
2019/02/01 01:07:53 [INFO] [addons] Setting up user addons
2019/02/01 01:07:53 [INFO] [addons] no user addons defined
2019/02/01 01:07:53 [INFO] Finished building Kubernetes cluster successfully
2019/02/01 01:07:53 [INFO] Restarting network, ingress, and metrics pods
2019/02/01 01:07:54 [INFO] Finished restoring snapshot [c-mc5hr-4rn9x] on all etcd hosts
2019/02/01 01:07:54 [INFO] kontainerdriver rancherkubernetesengine stopped
2019/02/01 01:07:54 [INFO] error in remotedialer server [400]: websocket: close 1006 unexpected EOF
2019/02/01 01:07:59 [INFO] kontainerdriver rancherkubernetesengine listening on address 127.0.0.1:10397
2019/02/01 01:07:59 [INFO] kontainerdriver rancherkubernetesengine stopped
2019/02/01 01:07:59 [INFO] Provisioning cluster [c-mc5hr]
2019/02/01 01:07:59 [INFO] Restoring cluster [c-mc5hr] from backup
2019/02/01 01:08:04 [INFO] kontainerdriver rancherkubernetesengine listening on address 127.0.0.1:10499
2019/02/01 01:08:04 [INFO] Restoring etcd snapshot c-mc5hr-4rn9x
2019/02/01 01:08:04 [INFO] Successfully Deployed state file at [management-state/rke/rke-842068286/cluster.rkestate]
2019/02/01 01:08:04 [INFO] [dialer] Setup tunnel for host 
2019/02/01 01:08:04 [WARNING] Failed to set up SSH tunneling for host ]: Can't establish dialer connection: can not build dialer to c-mc5hr:m-r6ltd
2019/02/01 01:08:04 [WARNING] Removing host  from node lists
2019/02/01 01:08:04 [INFO] kontainerdriver rancherkubernetesengine stopped




```

**Comment by moelsayed** [comment]
A regression caused by a fix for #17521, reverted for now.

**Comment by soumyalj** [comment]
Tested with master build from Feb 4th. etcd backup/restore is functional on a single node cluster with no data loss.
